### PR TITLE
Message meta

### PR DESCRIPTION
**Pull request by tusharmndr**

### CODE REVIEW

File: memq-actor/src/main/java/io/appform/memq/ActorSystem.java [hunk 1]
```diff
@@ -6,18 +6,17 @@
...
 import lombok.val;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.ExecutorService;
-import java.util.function.BiConsumer;
-import java.util.function.Consumer;
-import java.util.function.Function;
-import java.util.function.ToIntFunction;
+import java.util.function.*;
```
Comment (santanusinha) [inline memq-actor/src/main/java/io/appform/memq/ActorSystem.java:19]: revert

File: memq-actor/src/main/java/io/appform/memq/actor/InternalMessage.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package io.appform.memq.actor;
+
+import lombok.Value;
+
+//package private class
```
Comment (santanusinha) [inline memq-actor/src/main/java/io/appform/memq/actor/InternalMessage.java:5]: what is this?

File: memq-actor/src/main/java/io/appform/memq/actor/MessageMeta.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+
+@Getter
+@AllArgsConstructor
+public class MessageMeta {
+
+    private int deliveryAttempt;
```
Comment (santanusinha) [inline memq-actor/src/main/java/io/appform/memq/actor/MessageMeta.java:11]: make this volatile/atomic for safety

**Review by santanusinha — CHANGES_REQUESTED** [review]

File: memq-actor/src/main/java/io/appform/memq/ActorSystem.java (hunk 1 above)
Comment (tusharmndr) [reply memq-actor/src/main/java/io/appform/memq/ActorSystem.java:19]: Done

File: memq-actor/src/main/java/io/appform/memq/actor/InternalMessage.java (hunk 2 above)
Comment (tusharmndr) [reply memq-actor/src/main/java/io/appform/memq/actor/InternalMessage.java:7]: Removed

File: memq-actor/src/main/java/io/appform/memq/actor/MessageMeta.java (hunk 3 above)
Comment (tusharmndr) [reply memq-actor/src/main/java/io/appform/memq/actor/MessageMeta.java:13]: Done

**Review by santanusinha — APPROVED** [review]

**Pull request merged**